src/utils.cpp

```diff
@@ -195,6 +199,140 @@ bool is_lit(const Term & l, const Sort & boolsort)
   return false;
 }
 
+void cnf_to_dimacs(Term cnf, std::ostringstream & y)
+{
+  Sort sort = cnf->get_sort();
+  assert(sort->get_sort_kind() == BOOL);
+  if (cnf->is_value() && cnf->to_string() == "true")
+  {  // empty cnf formula
+    y << "p cnf 0 0\n";
+    return;
+  }
+  TermVec before_and_elimination({ cnf });
+  TermVec after_and_elimination;
+  // This while loop separate the clauses, after_and_elimination will contain all clauses
+  // because every smt::and op will be eliminated. This happens because until
+  // smt::and op is not detected that term is added back to before_and_elimination, and as no term
+  // with smt::or as the primOp is touched all clauses shall be separated and be
+  // intact
+  while (!before_and_elimination.empty())
+  {
+    Term t = before_and_elimination.back();
+    before_and_elimination.pop_back();
+    smt::Op op = t->get_op();
+    std::string ops = op.to_string();
+    assert(ops=="null" || ops=="or" || ops=="and" || ops=="not");
```

Comment:
No reason to use strings here, right? You can do it similarly to the line below.